XCUITests/FxScreenGraph.swift

```diff
@@ -452,6 +465,35 @@ func createScreenGraph(for test: XCTestCase, with app: XCUIApplication) -> MMScr
         screenState.backAction = navigationControllerBackAction
     }
 
+    map.addScreenState(FxASigninScreen) { screenState in
+        screenState.backAction = navigationControllerBackAction
+
+        screenState.gesture(forAction: Action.TypeInvalidEmail) { userState in
+            app.webViews.textFields["Email"].tap()
+            typeInField(userState.fxaWrongEmail)
+        }
+        screenState.gesture(forAction: Action.TypeCorrectEmail) { userState in
+            app.webViews.textFields["Email"].tap()
+            typeInField(userState.fxaRightEmail)
+        }
```

Comment:
There is a lot of duplication here. How can this be parameterized?

Suggest you strive for

```swift
userState.fxaEmail = "bad@email.com"
userState.fxaPassword = "password"
navigator.performAction(Action.FxATypeEmail)
navigator.performAction(Action.FxATypePassword)
```


```swift
userState.fxaEmail = "valid@email.com"
userState.fxaPassword = "password"
navigator.performAction(Action.FxATypeEmail)
navigator.performAction(Action.FxATypePassword)
navigator.performAction(Action.FxATapSignInButton)

// short cut, which logs you in.
navigator.performAction(Action.FxAValidLogin)

```

Reply:
Oh I see, thanks for the suggestion, the tests are clearer now and there is not such duplication in FxScreenGraph.